controllers/etcd/reconciler.go

```diff
@@ -280,6 +279,14 @@ func (r *Reconciler) delete(ctx context.Context, etcd *druidv1alpha1.Etcd) (ctrl
 		}, err
 	}
 
+	roleBindingValues := componentrolebinding.GenerateValues(etcd)
```

Comment:
Just a thought: the order of deletion of resources seems to be the same as order of creation of resources, but maybe inverting it makes more sense semantically. This isn't applicable only for rolebinding, but for all the objects created and managed by etcd controller.

Anyway in execution, we don't need to worry since this simply sets deletion timestamp on the objects and deletion of an object occurs only when all dependent objects are deleted. Maybe something we can improve upon when re-looking at each controller's code.

Reply:
Sure Shreyas